dm/ctl/master/start_task.go

```diff
@@ -48,6 +49,29 @@ func startTaskFunc(cmd *cobra.Command, _ []string) error {
 		return err
 	}
 
+	// If task's target db is configured with tls certificate related content
+	// the contents of the certificate need to be read and transferred to the dm-master
+	task := config.NewTaskConfig()
+	yamlErr := task.Decode(string(content))
+	if yamlErr != nil {
+		return yamlErr
+	}
+	if task.TargetDB.Security != nil {
+		loadErr := task.TargetDB.Security.LoadTLSContent()
+		if loadErr != nil {
+			return loadErr
+		}
+		// After decode and encode the task configuration yaml, we need to clear some duplicate configurations
+		// otherwise some error will occur such as:
+		// ErrConfigSyncerCfgConflict/ErrConfigLoaderCfgConflict/ErrConfigMydumperCfgConflict
+		for idx := range task.MySQLInstances {
```

Comment:
maybe you can write a RawDecode (decode not adjust) to avoid this. I'm afraid more field other than `task.MySQLInstances[idx].Syncer` should be cleared but we are not aware of them.

And if so, check task.TargetDB is not nil

Reply:
fixed  in https://github.com/pingcap/dm/pull/1910/commits/a2cea6e23fdfa9dcc86088a34fc6b95478565b51